### DISCUSSION TITLE
[General Issue] Dependence on dateutil

### DISCUSSION THREAD

**Discussion by ghost** [aridevelopment-de/datetimeparser#186] [Ideas]
## Make datetimeparser independent

Solve the library's dependency on dateutil, by building an own function for calculating relativedatetimes.

**Comment by Ari24-cb24** [comment]
Why do we have to make datetimeparser independent? What benefits do we gain from doing that?

**Comment by ghost** [comment]
Nothing in terms of code. But it would help us to differentiate our library more from others.

**Comment by Ari24-cb24** [comment]
I don't really see the point in doing that then. Our job is to parse strings into datetime objects, not to re-create an already existing library. If there's anything we can gain from building our own function for calculating relative datetimes code-wise then I would say: Go for it. But why re-inventing the wheel again? Apart from that, integrating already existing libraries is not something bad. Quite the opposite, as many people already use dateutil, it might be easier for them to integrate our software into their code. But that is not really a strong argument as that highly depends on our api.

Maybe you can provide another good reason, open for anything.
